testbed/tests/window/test_window.py

```diff
@@ -516,47 +895,96 @@ async def test_move_and_resize(second_window, second_window_probe):
             )
         ],
     )
-    async def test_full_screen(second_window, second_window_probe):
-        """Window can be made full screen"""
-        assert not second_window_probe.is_full_screen
+    async def test_window_state_same_as_current_without_intermediate_states(
+        app_probe, second_window, second_window_probe, state
+    ):
+        """Setting window state the same as current without any intermediate states is
+        a no-op and there should be no expected delay from the OS."""
+        if state == WindowState.MINIMIZED and not second_window_probe.supports_minimize:
+            pytest.xfail(
+                "This backend doesn't reliably support minimized window state."
+            )
+
+        second_window.content = toga.Box(style=Pack(background_color=CORNFLOWERBLUE))
+        second_window.show()
+        # Add delay to ensure windows are visible after animation.
+        await second_window_probe.wait_for_window("Secondary window is shown")
+
+        # Set the window state:
+        second_window.state = state
+        # Add delay to ensure windows are visible after animation.
+        await second_window_probe.wait_for_window(
+            f"Secondary window is in {state}",
+            minimize=True if state == WindowState.MINIMIZED else False,
+            full_screen=True if state == WindowState.FULLSCREEN else False,
+        )
+        assert second_window.state == state
+
+        # Set the window state the same as current:
+        second_window.state = state
+        # No need to wait for OS delay as the above operation should be a no-op.
```

Comment:
That's a load bearing "should" :-) 

If the implementation is incorrect, the "should" might not be correct.

Reply:
Same state transitions are now being tested at the core level.